### DISCUSSION TITLE
_app/index.tsx has "/index" route

### DISCUSSION THREAD

**Discussion by wonu** [kiliman/remix-flat-routes#37] [Q&A] [answered]
I tried as written in this [comment](https://github.com/remix-run/remix/discussions/4482#discussioncomment-4742179) and got a result somewhat different (on Windows, Remix `v1.11.1`, remix-flat-routes `v0.5.7`):
```
app/routes
├── _app
│   └── index.tsx
└── _app._index
    └── index.tsx
```
```tsx
$ npx remix routes
<Routes>
  <Route file="root.tsx">
    <Route path="index" file="routes\\_app\\index.tsx" />
    <Route index file="routes\\_app._index\\index.tsx" />
  </Route>
</Routes>
```

What am I missing?

**Comment by kiliman** [comment, marked as answer]
I believe there's a bug in the Remix `v2` routing.

`remix-flat-routes` treats this correctly.

`_app/index.tsx` is the layout route for `_app` (flat-folder) same as `_app.tsx`
`_app/_index/index.tsx` is the _index_ route ... same as `app._index.tsx`

If you use folders for organization, `remix-flat-routes` will assume you're using `flat-folders`, so it looks for the _route file_ from the immediate children. In `v2` convention, this can only be `index.tsx` or `route.tsx`. With `remix-flat-routes`, I've extended it so it can `index|page|route|layout` or *any* (tsx/jsx/mdx) file starting with a leading `_`.

For *index* routes, `v2` convention requires it to be named `_index`. An *index* route is the default route for a given path, like `/` or `/users`, etc.

With `remix-flat-routes` *index* routes can be named `index|_index`.

Hope this helps.

**Reply by wonu** [reply]
Thank you for your kind reply.

But I'm still confused, since I configured as `remix-flat-routes`.

Here's my project setting:
![remix-routes](https://user-images.githubusercontent.com/9602236/215954107-0d40c6d1-2dad-422b-b341-1a47f5fb4832.png)

Does the Remix `v2` routing still work even when not `v2_routeConvention`?

**Reply by kiliman** [reply]
Your configuration looks fine. `v2` and `remix-flat-routes` are separate things, so you should enable one or the other.

Anyway, I see you're on Windows, but using a _bash_ shell. That's something I haven't tried yet. It's probably messing with the file paths. Because if you look at the file prop, you'll see it's a Windows path, but I bet Node is seeing `path.sep = '/'`, which is what I use to determine folders.

**Reply by wonu** [reply]
I tried it with PowerShell but the result was same.

So I checked the code of this repo, and it seems I've found the cause, so created a PR(#40).
